healthvault-ios-sdk/Classes/Clients/MHVPersonClientProtocol.h

```diff
@@ -18,38 +18,72 @@
 
-- (void)setApplicationSettingsWithRequestParameters:(NSString *)requestParameters
-                                         completion:(void(^_Nullable)(NSError *_Nullable error))completion;
+/**
+ * Sets the application settings for the current application.
+ * The applicationSettings.appSettingsXml will not be processed by HealthVault, but must be valid XML
+ *
+ * @param completion Envoked when the operation completes.
+ *        NSError object will be nil if there is no error when performing the operation.
+ */
+- (void)setApplicationSettings:(MHVApplicationSettings *)settings
+                    completion:(void(^_Nullable)(NSError *_Nullable error))completion;
 
 /**
  * Gets information about people authorized for an application.
  *
- * @param completion Envoked when the operation completes. NSArray<MHVPersonInfo *> an array of MHVPersonInfo objects representing people authorized for the application. NSError object will be nil if there is no error when performing the operation.
+ * @param completion Envoked when the operation completes.
+ *        MHVGetAuthorizedPeopleResult containing an array of MHVPersonInfo objects representing people authorized for the application, and a moreResults flag.
+ *        NSError object will be nil if there is no error when performing the operation.
  */
-- (void)getAuthorizedPeopleWithCompletion:(void(^)(NSArray<MHVPersonInfo *> *_Nullable personInfos, NSError *_Nullable error))completion;
-
+- (void)getAuthorizedPeopleWithCompletion:(void(^)(MHVGetAuthorizedPeopleResult *_Nullable authorizedPeople, NSError *_Nullable error))completion;
 
 /**
  * Gets information about people authorized for an application.
  *
  * @param settings The MHVGetAuthorizedPeopleSettings object used to configure the results returned by this method.
- * @param completion Envoked when the operation completes. NSArray<MHVPersonInfo *> an array of MHVPersonInfo objects representing people authorized for the application. NSError object will be nil if there is no error when performing the operation.
+ *        This allows continuing if a previous getAuthorizedPeople call had the moreResults flag set
+ * @param completion Envoked when the operation completes. 
+ *        MHVGetAuthorizedPeopleResult containing an array of MHVPersonInfo objects representing people authorized for the application, and a moreResults flag.
+ *        NSError object will be nil if there is no error when performing the operation.
  */
-- (void)getAuthorizedPeopleWithSettings:(MHVGetAuthorizedPeopleSettings *_Nonnull)settings
-                             completion:(void(^)(NSArray<MHVPersonInfo *> *_Nullable personInfos, NSError *_Nullable error))completion;
+- (void)getAuthorizedPeopleWithSettings:(MHVGetAuthorizedPeopleSettings *)settings
+                             completion:(void(^)(MHVGetAuthorizedPeopleResult *_Nullable authorizedPeople, NSError *_Nullable error))completion;
```

Comment:
I don't think we should change the return type from and array of MHVPersonInfos to the MHVGetAuthorizedPeopleResult. We need to handle the case where there are "more results" internally, and not expose this internal object to the caller.